lib/http2.c

```diff
@@ -2011,6 +2012,20 @@ static ssize_t h2_submit(struct stream_ctx **pstream,
     }
   }
 
+  body = (const char *)buf + nwritten;
+  bodylen = len - nwritten;
+
+  if(bodylen > 0) {
+    /* We have request body to send in DATA frame */
+    ssize_t n = Curl_bufq_write(&stream->sendbuf, body, bodylen, err);
```

Comment:
Can this run into the bufq's max size? If not, I think the comment could mention this fact.

Reply:
In what way does it matter? Not sure i understand.

Reply:
Oh right, it will deal with the number of bytes that it can and return that amount so it does not matter. Sorry, I did not think this through until now. Case closed. Thanks!

Reply:
I believe it should not be possible since bufq should be empty at this point and the length of body cannot exceed `MAX_INITIAL_POST_SIZE`, which has the same value as `H2_STREAM_SEND_CHUNKS` * `H2_CHUNK_SIZE` (the max capacity of `stream->sendbuf`). However, since this fact is relying on different constant values that may or may not change in the future, I think maybe we should handle `CURLE_AGAIN` case by defer the DATA frame. What do you think?

Reply:
@peppapighs you accounted for such things by only incrementing `nwritten` by the amount acutally added to the `bufq`. The caller of the `send()` method has to be prepared that not all was written, as the same could happen to a HTTP/1.1 write onto a TCP socket.

Reply:
Oh right. That makes sense.